src/Router.sol

```diff
@@ -198,6 +202,28 @@ contract Router is IRouter, EIP712, FeeGenerator {
         _execute(user, logics, tokensReturn, referralCode);
     }
 
+    /// @notice Execute permitted execution data through the given user's agent. Creates an agent for user if not created.
+    ///         Fees are charged in the user's agent based on the scenarios defined in the FeeGenerator contract, which
+    ///         calculates fees by logics and msg.value.
+    /// @param details The execution details permitted by user
+    /// @param user The user address
+    /// @param signature The user's signature bytes
+    function executeBySig(
+        IParam.ExecutionDetails calldata details,
+        address user,
+        bytes calldata signature
+    ) external payable whenReady(user) {
+        uint256 deadline = details.deadline;
+        uint256 nonce = details.nonce;
+        // Verify deadline, signature and nonce
+        if (block.timestamp > deadline) revert SignatureExpired(deadline);
```

Comment:
Is it a good idea that uses a modifier for checking `deadline`?

Reply:
Since that there are two kinds of "details" here, we need to have multiple modifiers to handle them. I don't think that it will look better on the entire contract.